File: New folder/cpp/stack.cpp
#include <bits/stdc++.h>
#define ll long long
#define ld long double
#define forLoop(n) for(int i=0;i<n;i++)
#define space ' '
#define newline endl

using namespace std;

class node{
    public:
        int data;
        node *next;
        node(int data,node *next=NULL):data{data},next{next}{}
};

class lStack{
    int size;
    int top;
    int *arr;
    public:
        lStack(int decsize):arr{new int[decsize]},size{decsize},top{0}{}
        void pushEle(int data);
        void popEle();
        void peek();
};

void lStack::pushEle(int data){
    if(top==size){
        cout << "Stack overflow error" << newline;
        return;
    }
    arr[top++]=data;
    cout << "Pushed element " << data << newline;
}

void lStack::popEle(){
    if(top==0){
        cout << "Stack underflow error" << newline;
        return ;
    }
    cout << "Popped element " << arr[--top] << newline;
    return;
}

void lStack::peek(){
    if(top==0){
        cout << "EMPTY" << newline;
        return;
    }
    cout << arr[top-1] << newline;
}


class llStack{
    node * head;
    int top;
    int size;
    public:
        // constructor
        llStack(int size):size{size},head{NULL},top{0}{}

        // modifiers
        void pushEle(int data);
        void popEle();
        void peek();
};

void llStack::pushEle(int data){
    if(top==size){
        cout << "Stack overflow error" << newline;
        return;
    }
    auto toInsert=new node(data,head);
    head=toInsert;
    top+=1;
    cout << "Pushed element " << data << newline;
    return;
}

void llStack::popEle(){
    if(top==0){
        cout << "Stack underflow error" << newline;
        return;
    }
    auto temp=head;
    cout << "Popped element " << temp->data << newline;
    head=head->next;
    top-=1;
    delete temp;
}

void llStack::peek(){
    if(top==0){
        cout << "EMPTY" << newline;
        return;
    }
    cout << head->data << newline;
    return;
}

int main()
{
    std::ios_base::sync_with_stdio(false);
    std::cin.tie(NULL);

    cout << "USING ARRAY" << newline;
    lStack l1(5);
    l1.pushEle(4);
    l1.pushEle(5);
    l1.popEle();
    l1.peek();

    cout << "LINKED LIST" << newline;
    llStack l2(4);
    l2.pushEle(4);
    l2.pushEle(3);
    l2.pushEle(5);
    l2.popEle();
    l2.peek();
    return 0;
}

File: New folder/python/stack.py
from typing import Sized


class node:
    def __init__(self,data=None,next=None):
        self.data=data
        self.next=next

class stack:
    def __init__(self,size):
        self.head=node()
        self.size=size
        self.top=0

    def pushEle(self,data):
        if self.top==self.size: 
            print("Stack overflow error")
            return 
        print("Pushed element",data)
        if self.head is None:
            self.head.data=data 
            self.top+=1 
            return

        toInsert=node(data,self.head)
        self.top+=1
        self.head=toInsert

    def popEle(self):
        if self.top==0: 
            print("Stack underflow error")
            return 
        print("Popped element",self.head.data)
        toDel=self.head 
        self.top-=1
        self.head=self.head.next
        del toDel 

    def traverse(self):
        temp = self.head
        if self.top==0:
            print("Empty stack")
            return
        for i in range(self.top):
            print(temp.data,end=' ')
            temp=temp.next 
        print()

    def peek(self):
        if self.top==0:
            print("EMPTY")
            return 
        print(self.head.data)
    
        
if __name__=="__main__":
    value = stack(5)
    value.pushEle(4)
    value.pushEle(3)
    value.pushEle(2)
    value.traverse()
    value.popEle()
    value.popEle()
    value.popEle()
    value.popEle()
    value.peek()
